### PR TITLE
Machine ID Docs Refactor: Add `tctl` and `terraform` access guides

### PR DESCRIPTION
**Pull request by strideynet**
Adds content for the `tctl` and `terraform` access guides based on the new structure.

### CODE REVIEW

File: docs/pages/machine-id/access-guides/tctl.mdx [hunk 1]
```diff
@@ -3,4 +3,124 @@ title: Machine ID with `tctl`
 description: How to use Machine ID with `tctl` to manage your Teleport configuration
```
Comment (ptgott) [inline docs/pages/machine-id/access-guides/tctl.mdx:3]: The docs engine doesn't process backticks in titles and descriptions, so we should remove these from around "\`tctl\`"

File: docs/pages/machine-id/access-guides/tctl.mdx [hunk 2]
```diff
@@ -3,4 +3,124 @@ title: Machine ID with `tctl`
...
 ---
 
-TODO
\ No newline at end of file
+`tctl` is the Teleport cluster management CLI tool. Whilst it usually uses the
+credentials from the locally logged in user, it is also possible to use
+Machine ID credentials. This allows `tctl` to be leveraged as part of a custom
+automation deployed in a non-interactive environment (e.g CI/CD).
```
Comment (ptgott) [inline docs/pages/machine-id/access-guides/tctl.mdx:9]:
```suggestion
automation workflow deployed in a non-interactive environment (e.g CI/CD).
```
To agree with the article "a"

File: docs/pages/machine-id/access-guides/tctl.mdx [hunk 3]
```diff
@@ -3,4 +3,124 @@ title: Machine ID with `tctl`
...
+
+Run `tctl`, replacing `example.teleport.sh:443` with the address of your
+Teleport Proxy or Auth Server and `/opt/machine-id/identity` with the path to
+the generated identity file if you have modified this:
+
+```code
+$ tctl --auth-server example.teleport.sh:443 -i /opt/machine-id/identity create -f roles/*.yaml
+```
+
+Check your Teleport cluster, ensuring the role has been created.
```
[the remark this answers is not in the document]
Comment (strideynet) [reply docs/pages/machine-id/access-guides/tctl.mdx:121]: Agreed

File: docs/pages/machine-id/access-guides/tctl.mdx (hunk 1 above)
Comment (strideynet) [reply docs/pages/machine-id/access-guides/tctl.mdx:3]: 93807c7e38

File: docs/pages/machine-id/access-guides/tctl.mdx [hunk 4]
```diff
@@ -3,4 +3,124 @@ title: Machine ID with `tctl`
...
+```yaml
+kind: role
+version: v6
+metadata:
+  name: example-role
+spec:
+  allow:
+    logins: ['root']
+    node_labels:
+      'key': 'value'
```
[the remark this answers is not in the document]
Comment (strideynet) [reply docs/pages/machine-id/access-guides/tctl.mdx]: 9a0e6e0936

File: docs/pages/machine-id/access-guides/terraform.mdx [hunk 5]
```diff
@@ -3,4 +3,177 @@ title: Machine ID with the Teleport Terraform Provider
...
+    description = "Example role created by Terraform"
+  }
+
+  spec = {
+    allow = {
+      logins = ["root"]
+      node_labels = {
+        key    = ["value"]
+      }
+    }
```
[the remark this answers is not in the document]
Comment (strideynet) [reply docs/pages/machine-id/access-guides/terraform.mdx]: 9a0e6e0936

File: docs/pages/machine-id/access-guides/tctl.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (strideynet) [reply docs/pages/machine-id/access-guides/tctl.mdx:119]: a99e204ba0

File: docs/pages/machine-id/access-guides/terraform.mdx [hunk 6]
```diff
@@ -3,4 +3,177 @@ title: Machine ID with the Teleport Terraform Provider
...
+- `tbot` must already be installed and configured on the machine that will
+  run Terraform. For more information, see the
+  [platform guides](../platform-guides.mdx).
+
+## Step 1/3. Configure RBAC
+
+First, Teleport must be configured to allow the credentials produced by the bot
+to modify the Teleport configuration.
+
+If you have followed an access guide, you will have created a role and granted
```
[the remark this answers is not in the document]
Comment (strideynet) [reply docs/pages/machine-id/access-guides/terraform.mdx:35]: a99e204ba0

File: docs/pages/machine-id/access-guides/tctl.mdx [hunk 7]
```diff
@@ -3,4 +3,124 @@ title: Machine ID with `tctl`
...
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- (!docs/pages/includes/tctl.mdx!)
+- `tbot` must already be installed and configured on the machine that will
+  use `tctl`. For more information, see the
+  [platform guides](../platform-guides.mdx).
+
+## Step 1/3. Configure RBAC
+
+First, Teleport must be configured to allow the credentials produced by the bot
```
[the remark this answers is not in the document]
Comment (strideynet) [reply docs/pages/machine-id/access-guides/tctl.mdx]: [9e68d97](https://github.com/gravitational/teleport/pull/32036/commits/9e68d97ba132b467bb28664bde93db952828a10c)

**Pull request merged**